packages/dd-trace/src/llmobs/sdk.js

```diff
@@ -0,0 +1,448 @@
+
+      llmobs._tagger.setLLMObsSpanTags(span, valid && kind, {
+        ...llmobsOptions,
+        parentLLMObsSpan
+      })
+      llmobs.annotate(span, { inputData: getFunctionArguments(fn, arguments) })
+
+      try {
+        const result = fn.apply(this, arguments)
+
+        if (result && typeof result.then === 'function') {
+          return result.then(value => {
+            if (value && kind !== 'retrieval' && !LLMObsTagger.tagMap.get(span)?.[OUTPUT_VALUE]) {
+              llmobs.annotate(span, { outputData: value })
+            }
+            if (llmobs.enabled) storage.enterWith(oldStore)
+            return value
+          }).catch(err => {
+            if (llmobs.enabled) storage.enterWith(oldStore)
+            throw err
+          })
+        }
+
+        if (result && kind !== 'retrieval' && !LLMObsTagger.tagMap.get(span)?.[OUTPUT_VALUE]) {
+          llmobs.annotate(span, { outputData: result })
+          if (llmobs.enabled) storage.enterWith(oldStore)
+        }
+
+        return result
+      } catch (e) {
+        if (llmobs.enabled) storage.enterWith(oldStore)
+        throw e
+      }
+    }
+
+    return this._tracer.wrap(name, spanOptions, wrapped)
+  }
+
+  annotate (span, options) {
+    if (!this.enabled) {
+      logger.warn(
+        'Annotate called while LLMObs is disabled. Not annotating span.'
+      )
+      return
+    }
+
+    if (!span) {
+      span = this.active()
+    }
+
+    if ((span && !options) && !(span instanceof Span)) {
+      options = span
+      span = this.active()
+    }
+
+    if (!span) {
+      logger.warn('No span provided and no active LLMObs-generated span found')
+      return
+    }
+    if (!options) {
+      logger.warn('No options provided for annotation.')
+      return
+    }
+
+    if (!LLMObsTagger.tagMap.has(span)) {
+      logger.warn('Span must be an LLMObs-generated span')
+      return
+    }
+    if (span._duration !== undefined) {
+      logger.warn('Cannot annotate a finished span')
+      return
+    }
+
+    const spanKind = LLMObsTagger.tagMap.get(span)[SPAN_KIND]
+    if (!spanKind) {
+      logger.warn('LLMObs span must have a span kind specified')
+      return
+    }
+
+    const { inputData, outputData, metadata, metrics, tags } = options
+
+    if (inputData || outputData) {
+      if (spanKind === 'llm') {
+        this._tagger.tagLLMIO(span, inputData, outputData)
+      } else if (spanKind === 'embedding') {
+        this._tagger.tagEmbeddingIO(span, inputData, outputData)
+      } else if (spanKind === 'retrieval') {
+        this._tagger.tagRetrievalIO(span, inputData, outputData)
+      } else {
+        this._tagger.tagTextIO(span, inputData, outputData)
+      }
+    }
+
+    if (metadata) {
+      this._tagger.tagMetadata(span, metadata)
+    }
+
+    if (metrics) {
+      this._tagger.tagMetrics(span, metrics)
+    }
+
+    if (tags) {
+      this._tagger.tagSpanTags(span, tags)
+    }
+  }
+
+  exportSpan (span) {
+    if (!this.enabled) {
```

Comment:
I believe we can remove this check for self.enabled completely - i don't think llmobs being enabled/disabled should change the behavior of export_span, which should just return the `trace/span id` of currently active llm obs span

Reply:
Since this function returns `undefined` in other cases, i'd assume the user has some `undefined` handling implemented in their code if they use export span. Regardless, I think it's good to minimize the behavior difference in public api methods between enabled/disabled llmobs

Reply:
ah ok, i had lifted it right from [this](https://github.com/DataDog/dd-trace-py/blob/2a4d8edf2ed53f203b2edd82ae047be892df5dbc/ddtrace/llmobs/_llmobs.py#L795-L799). but, i also turn off context tracking if llmobs is disabled (so we can properly track parentage) - [example](https://github.com/DataDog/dd-trace-js/blob/7c7a328b2f52da0b9b515ca5d32e1af23bdf3097/packages/dd-trace/src/llmobs/sdk.js#L186) (`enterWith` is what's tracking this context). This just means if LLMObs is disabled, `exportSpan` will always return undefined, so it'll just trigger the check below this one, so it should be safe to remove if we wanna have less of a distinction for enabled/disabled LLMObs here

Reply:
let me know if that flow makes sense to you!